app/views/carts/show.html.erb

```diff
@@ -12,4 +10,6 @@
   </div>
 <% end  %>
 <p>Grand Total: <%= number_to_currency(@cart.grand_total/100) %></p>
-<%= link_to "Empty Cart", cart_path %>
+  <% unless @cart_items.count == 0 %>
```

Comment:
I'm not sure, but I think this is the kind of logic we're supposed to avoid in the View, where you're actually asking the view to do some work(`count`). It's not a database call though so I'm not really sure. Either way, I created a `cart_count` method for my user story (it's in master now) that you could use here - `@cart.cart_count`!